aggregator_core/src/datastore/tests.rs

```diff
@@ -1794,95 +1795,189 @@ async fn aggregation_job_acquire_release(ephemeral_datastore: EphemeralDatastore
+                tx.put_aggregation_job(&AggregationJob::<
+                    VERIFY_KEY_LENGTH,
+                    TimeInterval,
+                    Prio3Count,
+                >::new(
                     *task.id(),
-                    random(),
+                    finished_aggregation_job_id,
                     (),
                     (),
                     Interval::new(OLDEST_ALLOWED_REPORT_TIMESTAMP, Duration::from_seconds(1))
                         .unwrap(),
                     AggregationJobState::Finished,
                     AggregationJobStep::from(1),
-                ),
-            )
-            .await
-            .unwrap();
+                ))
+                .await
+                .unwrap();
 
-            // Write an expired aggregation job. We don't want to retrieve this one, either.
-            tx.put_aggregation_job(
-                &AggregationJob::<VERIFY_KEY_LENGTH, TimeInterval, Prio3Count>::new(
+                // Write an expired aggregation job. We don't want to retrieve this one, either.
+                let expired_aggregation_job_id = random();
+                tx.put_aggregation_job(&AggregationJob::<
+                    VERIFY_KEY_LENGTH,
+                    TimeInterval,
+                    Prio3Count,
+                >::new(
                     *task.id(),
-                    random(),
+                    expired_aggregation_job_id,
                     (),
                     (),
                     Interval::new(Time::from_seconds_since_epoch(0), Duration::from_seconds(1))
                         .unwrap(),
                     AggregationJobState::InProgress,
                     AggregationJobStep::from(0),
-                ),
-            )
-            .await
-            .unwrap();
+                ))
+                .await
+                .unwrap();
 
-            // Write an aggregation job for a task that we are taking on the helper role for.
-            // We don't want to retrieve this one, either.
-            let helper_task =
-                TaskBuilder::new(task::QueryType::TimeInterval, VdafInstance::Prio3Count)
-                    .build()
-                    .helper_view()
-                    .unwrap();
-            tx.put_aggregator_task(&helper_task).await.unwrap();
-            tx.put_aggregation_job(
-                &AggregationJob::<VERIFY_KEY_LENGTH, TimeInterval, Prio3Count>::new(
+                // Write an aggregation job for a task that we are taking on the helper role for.
+                // We don't want to retrieve this one, either.
+                let helper_task =
+                    TaskBuilder::new(task::QueryType::TimeInterval, VdafInstance::Prio3Count)
+                        .build()
+                        .helper_view()
+                        .unwrap();
+                tx.put_aggregator_task(&helper_task).await.unwrap();
+                let helper_aggregation_job_id = random();
```

Comment:
nit: This wasn't necessary to return / is otherwise unused, so this could have remained an inline `random()` call.